Replace function calls with properties where possible for SCVMM refresh.

djberg96 opened the pull request:
This is an interim improvement for the SCVMM inventory collection script. It turns out that some of the information we were collecting for VM's and images using powershell functions in a loop were already available as properties. This means a 3x reduction in the number of calls per VM and a 2x reduction in the number of calls per image.

While not as fast as the upcoming JSON implementation, I saw significant improvements. As a standalone WinRM script, the older implementation typically took 120-150 seconds. This implementation typically came in around 80 seconds.

djberg96 commented:
@bronaghs Everything looked good in the UI, but please review and make sure I didn't miss or break anything.

chessbyte commented:
Nice find!
/cc @blomquisg

Fryguy commented:
@djberg96 Part of the reason we had single spaces was because there is a maximum size to the script that could be passed down.  Is that still a problem?

djberg96 commented:
@Fryguy True, I added some very minor formatting, but the overall number of bytes has been reduced. I modified the functions to use inline parameters, and shrunk some variable names, in addition to the removal of the function calls.

Fryguy commented:
Sounds good to me :+1:

bronaghs commented:
I didnt have a chance to review this yesterday afternoon before it was merged, this is a really nice improvement. Thanks @djberg96
